pkg/bmc/idrac_virtualmedia.go

```diff
@@ -90,3 +92,52 @@ func (a *redfishiDracVirtualMediaAccessDetails) VendorInterface() string {
 func (a *redfishiDracVirtualMediaAccessDetails) SupportsSecureBoot() bool {
 	return true
 }
+
+func (a *redfishiDracVirtualMediaAccessDetails) ValidateBootInterface(bmcCreds Credentials) (bool, error) {
+	bmcConfig := gofish.ClientConfig{
+		Endpoint: getRedfishAddress(a.bmcType, a.host),
+		Username: bmcCreds.Username,
+		Password: bmcCreds.Password,
+		Insecure: a.disableCertificateVerification,
+	}
+
+	client, err := gofish.Connect(bmcConfig)
+	if err != nil {
+		return false, err
+	}
+	defer client.Logout()
+
+	service := client.Service
+
+	managers, err := service.Managers()
+	if err != nil {
+		return false, err
+	}
+
+	for _, manager := range managers {
+		vmediaList, err := manager.VirtualMedia()
+		if err != nil {
+			return false, err
+		}
+		if vmediaList == nil {
+			continue
+		}
+
+		for _, vmedia := range vmediaList {
+			for _, media := range vmedia.MediaTypes {
+				if media != "CD" {
+					continue
+				}
+
+				if vmedia.SupportsMediaInsert && vmedia.SupportsMediaEject {
+					return true, nil
+				} else {
+					return false, &BootInterfaceValidationError{message: "VirtualMedia is missing one of the following Actions: InsertMedia, EjectMedia."}
```

Comment:
Is it possible for a host to have multiple devices that need to be checked?

Can the error report which action(s) are missing?

Why would an action be missing? Can the user do anything about the situation?

Reply:
Dell folks told me that we should check the CD device based on https://github.com/dell/iDRAC-Redfish-Scripting/blob/d3e674b05b7b699c4af185ebb9c4993a904e4128/Redfish%20Python/InsertEjectVirtualMediaREDFISH.py#L42-L55

Reply:
And there will only ever be one device with `media == "CD"`?

Reply:
This is a good question, In all Dell machines I've tested to access the BMC's there was only 1 device with CD.

Reply:
It makes sense for virtual devices, but I wasn't sure if it was safe to assume.

Reply:
yeah, I've tested in different Dell Hardware with idrac8 and idrac9